psa-crypto/src/operations/key_derivation.rs

```diff
@@ -0,0 +1,92 @@
+// Copyright 2020 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+
+//! # Key Derivation operations
+
+use super::key_management::complete_new_key_operation;
+use crate::initialized;
+use crate::types::key::Attributes;
+use crate::types::key::Id;
+use crate::types::key_derivation::Operation;
+use crate::types::status::{Result, Status};
+use core::convert::{TryFrom, TryInto};
+
+/// This function calculates output bytes from a key derivation algorithm and uses those bytes to generate a key deterministically.
+/// The key's location, usage policy, type and size are taken from attributes.
+/// # Example
+///
+/// ```
+/// use psa_crypto::operations::{key_derivation, key_management};
+/// use psa_crypto::types::key::{Attributes, Type, Lifetime, Policy, UsageFlags};
+/// use psa_crypto::types::algorithm::{Hash, KeyDerivation};
+/// use psa_crypto::types::key_derivation::{Operation, Inputs, Input};
+///
+/// # const KEY_DATA: [u8; 23] = [0; 23];
```

Comment:
In your example, could this data come from the key agreement algorithm?

Reply:
I think you'd use the output of a key agreement algorithm as an input to the key derivation step `secret` as an alternative to creating a new key to be used in the `secret` input step. Whether they result in the same outcome, I'm not sure.

Reply:
I think it would be nice for the example to show the key agreement + key derivation example!